### DISCUSSION TITLE
Count relationships

### DISCUSSION THREAD

**Discussion by teovillanueva** [prisma/prisma#2512] [Q&A] [answered]
Hi everyone, im having trouble counting relationships with my prisma client.

This is my schema
```
model User {
  id         Int       @default(autoincrement()) @id
  username   String    @unique
  followedBy User[]    @relation("UserFollows", references: [id])
  following  User[]    @relation("UserFollows", references: [id])
}
```

i can fetch the data successfully with this line
`const user = await prisma.user.findOne({ where: { id: 1 }, include: { following: true } });`
but is there a way to fetch the user data and count the `following` relationship?
if the user follows `n` other users, following should be `n`. I want to count the number of people the user is following without doing `user.following.length` or making 2 or more queries.

Thank you!

**Comment by ryands17** [comment]
Hey @teovillanueva :wave: 
Currently Prisma doesn't support `count` and fetching the data in a single query, so to get the data, you would have to add a separate call.
Could you open a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) adding the above use case as well?

**Reply by teovillanueva** [reply]
Thanks! i'll open a feauture request right now. Should i mark this as answer?

**Comment by teovillanueva** [comment, marked as answer]
This is the link to the [issue](https://github.com/prisma/prisma/issues/2519)

**Comment by ludralph** [comment]
Hi @teovillanueva,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
